File: js/tests/nameGreeting.test.js
const helloName = require("../nameGreeting");

const data = [
  ["Gerald", "Hello Gerald!"],
  ["Fatima", "Hello Fatima!"],
  ["Ed", "Hello Ed!"],
  ["Tiffany", "Hello Tiffany!"]
];

describe("helloName", () => {
  describe("Return a greeting from name input", () => {
    test.each(data)("Input: %s, Output: %s", (name, expected) => {
      var actual = helloName(name);

      expect(actual).toEqual(expected);
    });
  });
});

File: CSharp/Tests/NameGreetingTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class NameGreetingTest
    {
        [Theory]
        [InlineData("Gerald", "Hello Gerald!")]
        [InlineData("Fatima", "Hello Fatima!")]
        [InlineData("Ed", "Hello Ed!")]
        [InlineData("Tiffany", "Hello Tiffany!")]
        public void HelloName_StringValue_ReturnString(string name, string expected)
        {
            var actual = NameGreeting.HelloName(name);

            Assert.Equal(expected, actual);
        }
    }
}